### PR TITLE
Add Skip Posting settings

### PR DESCRIPTION
**Pull request by jaymckinney-enavate**
Added 'skip posting' config settings and incorporated them in the posting code to give the ability to skip auto posting of All, or Account, Customer, Bank, and/or Vendor batches.

### CODE REVIEW

File: Apps/W1/HybridGP/app/src/tables/GPCompanyAdditionalSettings.table.al [hunk 1]
```diff
@@ -320,6 +320,77 @@ table 40105 "GP Company Additional Settings"
...
             InitValue = true;
         }
+        field(35; "Skip All Posting"; Boolean)
+        {
+            InitValue = false;
+            DataClassification = SystemMetadata;
+
+            trigger OnValidate()
+            begin
+                if Rec."Skip All Posting" then begin
```
Comment (ToddE123) [inline Apps/W1/HybridGP/app/src/tables/GPCompanyAdditionalSettings.table.al:330]: Ah. The table sets the other Skip fields. If this is reflected in the UI then we are better. Can the user still change one of the other Skip values in the UI after this?

File: Apps/W1/HybridGP/app/src/Migration/Support/HelperFunctions.codeunit.al [hunk 2]
```diff
@@ -1220,10 +1221,12 @@ Codeunit 4037 "Helper Functions"
         StartTime := CurrentDateTime();
         Session.LogMessage('00007GJ', 'Posting GL transactions started.', Verbosity::Normal, DataClassification::SystemMetadata, TelemetryScope::ExtensionPublisher, 'Category', GetTelemetryCategory());
 
+        SkipPosting := GPCompanyAdditionalSettings.GetSkipAllPosting();
```
[the remark this answers is not in the document]
Comment (jaymckinney-enavate) [reply Apps/W1/HybridGP/app/src/Migration/Support/HelperFunctions.codeunit.al:1224]: The OnSkipHostingXXX procedures are events exposed for 3rd party PTEs. The config change allows for that functionality to continue to override as it does today.

File: Apps/W1/HybridGP/app/src/pages/GPMigrationConfiguration.Page.al [hunk 3]
```diff
@@ -209,6 +209,88 @@ page 4050 "GP Migration Configuration"
                 }
             }
 
+            group(SkipPosting)
```
[the remark this answers is not in the document]
Comment (jaymckinney-enavate) [reply Apps/W1/HybridGP/app/src/pages/GPMigrationConfiguration.Page.al:212]: The table handles this logic

File: Apps/W1/HybridGP/app/src/tables/GPCompanyAdditionalSettings.table.al (hunk 1 above)
Comment (jaymckinney-enavate) [reply Apps/W1/HybridGP/app/src/tables/GPCompanyAdditionalSettings.table.al:330]: Yes, the table handles this logic and the UI reacts accordingly.
When one of the other states change to false, the Skip All will then become false.

**Review by ToddE123 — APPROVED** [review]

**Pull request closed without merging**